app/shared/ui-foundation/src/commonMain/kotlin/ui/foundation/lists/PaginatedListState.kt

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (C) 2024-2025 OpenAni and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license, which can be found at the following link.
+ *
+ * https://github.com/open-ani/ani/blob/main/LICENSE
+ */
+
+package me.him188.ani.app.ui.foundation.lists
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+
+/**
+ * 分组数据接口，定义分组的基本属性
+ */
+interface GroupData {
+    val title: String
+}
+
+/**
+ * 可分页列表的分组配置
+ */
+interface PaginatedGroupConfig<T> {
+    /**
+     * 每个分组的最大条目数
+     */
+    val itemsPerGroup: Int
+
+    /**
+     * 检查某个条目是否属于指定的播放状态
+     */
+    fun isPlaying(item: T): Boolean = false
+
+    /**
+     * 生成指定分组的标题
+     */
+    fun generateGroupTitle(groupIndex: Int, itemsInGroup: List<T>): String
+}
+
+/**
+ * 默认的分组配置实现
+ */
+class DefaultPaginatedGroupConfig<T>(
+    override val itemsPerGroup: Int,
+    private val playingChecker: ((T) -> Boolean)? = null,
+    private val titleGenerator: (Int, List<T>) -> String,
+) : PaginatedGroupConfig<T> {
+    override fun isPlaying(item: T): Boolean = playingChecker?.invoke(item) ?: false
+    override fun generateGroupTitle(groupIndex: Int, itemsInGroup: List<T>): String =
+        titleGenerator(groupIndex, itemsInGroup)
+}
```

Comment:
Too complex, 这样就够了. 作为内部使用的东西, 并不需要保证极强的扩展性.

```suggestion
/**
 * 默认的分组配置实现
 */
class PaginatedGroupConfig<T>(
   val itemsPerGroup: Int,
   val getTitle: (item: T) -> String,
) : PaginatedGroupConfig<T>
```